packages/x-data-grid-pro/src/hooks/features/treeData/useGridTreeDataPreProcessors.tsx

```diff
@@ -204,6 +208,42 @@ export const useGridTreeDataPreProcessors = (
     [privateApiRef, props.disableChildrenSorting],
   );
 
+  const updateGridRowAriaAttributes = React.useCallback<GridPipeProcessor<'ariaAttributes'>>(
+    (attributes, rowId) => {
+      const rowNode = privateApiRef.current.getRowNode(rowId);
+      const ariaAttributes = attributes as Record<string, string | number | boolean>;
+
+      if (props.treeData !== true || rowNode === null) {
+        return ariaAttributes;
+      }
+
+      const filteredTopLevelRowCount = gridFilteredTopLevelRowCountSelector(privateApiRef);
+      const filteredDescendantCountLookup =
+        gridFilteredDescendantCountLookupSelector(privateApiRef);
+      const sortedVisibleRowPositionsLookup = gridExpandedSortedRowIdsLookupSelector(privateApiRef);
+
+      ariaAttributes['aria-level'] = rowNode.depth + 1;
+
+      const filteredDescendantCount = filteredDescendantCountLookup[rowNode.id] ?? 0;
+      // aria-expanded should only be added to the rows that contain children
+      if (rowNode.type === 'group' && filteredDescendantCount > 0) {
+        ariaAttributes['aria-expanded'] = Boolean(rowNode.childrenExpanded);
+      }
+
+      // if the parent is null, set size and position cannot be determined
+      if (rowNode.parent !== null) {
+        ariaAttributes['aria-setsize'] =
```

Comment:
Should `aria-setsize` only reflect the number of direct children?

<img width="1455" src="https://github.com/mui/mui-x/assets/13808724/3a870631-2ef9-4163-8b20-78e2fb5fddbb">

Reply:
that is correct
used children.length before, but that is not the same as the selector count 
I will update this and add a test to cover this case

Reply:
Updated. Added children count lookup to the existing filter state